### PR TITLE
[WIP] Fix appveyor tests

### LINKED ISSUE #1034: No tests on AppVeyor
It appears that since 07906673afbb7a4f606862ded2a02f1b7826124b tests have been disabled on AppVeyor.

Was that intended?

If not, could you maybe take a look at that @erthalion?

### PR DESCRIPTION
**Pull request by erthalion**
Fixes #1034

### CODE REVIEW

**Comment by erthalion** [comment]
Ok, apparently tests are included now, and one is failing :(

**Comment by Gabriel439** [comment]
The test failure appears to be due to the fact that this predicate fails to match on Windows because the Windows uses different path separators:

https://github.com/dhall-lang/dhall-haskell/blob/79e33b46f3252ca220553df0d3fc2e60e788a283/dhall/tests/Dhall/Test/Import.hs#L63-L70

That in turn means that the `XDG_CACHE_HOME` environment variable is not set, which leads to the import resolution failure in that test.

**Comment by Gabriel439** [comment]
@erthalion: If you merge https://github.com/erthalion/dhall-haskell/pull/2 into your branch then it should fix the Windows test failures

**Comment by sjakobi** [comment]
The test is still failing with

```
      .\dhall-lang\tests\import\success\hashFromCacheA.dhall:                                   FAIL
        Exception: 
        Error: No valid imports
        
        11: missing sha256:efc43103e49b56c5bf089db8e0365bbfc455b8a2f0dc6ee5727a3586f85969fd
        
        11:1
```

**Comment by sjakobi** [comment]
Oh, the test is actually fixed!

Now appveyor fails due to a compilation failure in `dhall-lsp-server`:

```
C:\projects\dhall-haskell\dhall-lsp-server\test\Backend\Dhall\DiagnosticsSpec.hs:14:40: error:
    Module
    `Dhall.LSP.Handlers.Diagnostics'
    does not export
    `compilerDiagnostics'
   |
14 | import Dhall.LSP.Handlers.Diagnostics (compilerDiagnostics)
   |                                        ^^^^^^^^^^^^^^^^^^^
```

cc @EggBaconAndSpam 

Note that the issue cannot be observed on Hydra as Hydra doesn't seem to build `dhall-lsp-server` at all!

**Comment by Gabriel439** [comment]
@sjakobi: Hydra does build `dhall-lsp-server` (specifically the binary tarball), but doesn't run the test suite.

You can add packages that you want to build test suites for here:

https://github.com/dhall-lang/dhall-haskell/blob/250fdfc34878a2103c00bc541fc47b4983cb41e6/release.nix#L63-L64

**Comment by Gabriel439** [comment]
@sjakobi: Also, I believe you can fix the `dhall-lsp-server` build failure by adding `compilerDiagnostics` to the export list of `Dhall.LSP.Handlers.Diagnostics`

**Comment by Gabriel439** [comment]
Actually, it looks like there is a reason why the Nix CI disables the test suite for `dhall-lsp-server`:

https://github.com/dhall-lang/dhall-haskell/blob/250fdfc34878a2103c00bc541fc47b4983cb41e6/nix/shared.nix#L129-L131

The Nix CI creates source distributions (a.k.a. "sdists") for each Cabal package before building them to ensure that they will be buildable from Hackage.  The source distribution for `dhall-lsp-server` does not contain the `dhall-lang` submodule, so the test suite would fail when built from Hackage.  That means that the test suite needs to be fixed to only depend on data-files local to the `dhall-lsp-server` package.

**Comment by sjakobi** [comment]
> Also, I believe you can fix the `dhall-lsp-server` build failure by adding `compilerDiagnostics` to the export list of `Dhall.LSP.Handlers.Diagnostics`

That fixes the build but then the tests fail:

```
test/Driver.hs
  prelude
    Backend.Dhall.Diagnostics
      should produce an empty diagnostics for an empty file:                 FAIL
        expected: []
         but got: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 0}, _end = Position {_line = 0, _character = 0}}, _severity = Just DsError, _code = Nothing, _source = Just "Dhall.Parser", _message = "unexpected end of input\nexpecting expression or whitespace\n", _relatedInformation = Nothing}]
      should produce an empty diagnostics for a correct file (sanity check): OK (0.03s)
      should produce correct diagnostics for various missing imports:        FAIL
        expected: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 0}, _end = Position {_line = 1, _character = 0}}, _severity = Just DsError, _code = Nothing, _source = Just "dhall-lsp-server", _message = "\nError: No valid imports", _relatedInformation = Nothing}]
         but got: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 0}, _end = Position {_line = 0, _character = 7}}, _severity = Just DsError, _code = Nothing, _source = Just "Dhall.Import", _message = "\n\ESC[1;31mError\ESC[0m: No valid imports", _relatedInformation = Nothing}]
      should produce correct diagnostic for various parser errors:           FAIL
        expected: [Diagnostic {_range = Range {_start = Position {_line = 5, _character = 8}, _end = Position {_line = 5, _character = 9}}, _severity = Just DsError, _code = Nothing, _source = Just "dhall-lsp-server", _message = "expecting label or whitespace\n", _relatedInformation = Nothing}]
         but got: [Diagnostic {_range = Range {_start = Position {_line = 5, _character = 8}, _end = Position {_line = 5, _character = 8}}, _severity = Just DsError, _code = Nothing, _source = Just "Dhall.Parser", _message = "expecting label or whitespace\n", _relatedInformation = Nothing}]
      should produce correct diagnostic for various typecheck errors:        FAIL
        expected: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 0}, _end = Position {_line = 1, _character = 0}}, _severity = Just DsError, _code = Nothing, _source = Just "dhall-lsp-server", _message = "Record mismatch", _relatedInformation = Nothing}]
         but got: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 0}, _end = Position {_line = 0, _character = 28}}, _severity = Just DsError, _code = Nothing, _source = Just "Dhall.TypeCheck", _message = "\n\ESC[1;31mError\ESC[0m: Record mismatch\n\n1\9474 { foo = 1 } \8743 { bar = Text }\n\ncombineMixedRecords.dhall:1:1", _relatedInformation = Nothing}]
      should produce correct diagnostic for various typecheck errors:        FAIL
        expected: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 15}, _end = Position {_line = 0, _character = 16}}, _severity = Just DsError, _code = Nothing, _source = Just "dhall-lsp-server", _message = "duplicate field: a\n", _relatedInformation = Nothing}]
         but got: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 15}, _end = Position {_line = 0, _character = 15}}, _severity = Just DsError, _code = Nothing, _source = Just "Dhall.Parser", _message = "duplicate field: a\n", _relatedInformation = Nothing}]
      should produce correct diagnostic for various typecheck errors:        FAIL (0.05s)
        expected: [Diagnostic {_range = Range {_start = Position {_line = 6, _character = 4}, _end = Position {_line = 49, _character = 0}}, _severity = Just DsError, _code = Nothing, _source = Just "dhall-lsp-server", _message = "\10096Sort\10097 has no type, kind, or sort", _relatedInformation = Nothing}]
         but got: [Diagnostic {_range = Range {_start = Position {_line = 6, _character = 4}, _end = Position {_line = 48, _character = 12}}, _severity = Just DsError, _code = Nothing, _source = Just "Dhall.TypeCheck", _message = "\n\ESC[1;31mError\ESC[0m: \10096Sort\10097 has no type, kind, or sort\n\n 7\9474     let U = \8704(X : Kind) \8594 (pow (pow X) \8594 X) \8594 pow (pow X)\n 8\9474 \n 9\9474 in  let tau\n10\9474         : pow (pow U) \8594 U\n11\9474         =   \955(t : pow (pow U))\n12\9474           \8594 \955(X : Kind)\n13\9474           \8594 \955(f : pow (pow X) \8594 X)\n14\9474           \8594 \955(p : pow X)\n15\9474           \8594 t (\955(x : U) \8594 p (f (x X f)))\n16\9474 \n17\9474 in  let sigma : U \8594 pow (pow U) = \955(s : U) \8594 s U (\955(t : pow (pow U)) \8594 tau t)\n18\9474 \n19\9474 in  let Delta\n20\9474         : pow U\n21\9474         = \955(y : U) \8594 not (\8704(p : pow U) \8594 sigma y p \8594 p (tau (sigma y)))\n22\9474 \n23\9474 in  let Omega : U = tau (\955(p : pow U) \8594 \8704(x : U) \8594 sigma x p \8594 p x)\n24\9474 \n25\9474 in  let Theta : Type = \8704(p : pow U) \8594 (\8704(x : U) \8594 sigma x p \8594 p x) \8594 p Omega\n26\9474 \n==================================================================================\n30\9474         : \8704(p : pow U) \8594 (\8704(x : U) \8594 sigma x p \8594 p x) \8594 p Omega\n31\9474         =   \955(p : pow U)\n32\9474           \8594 \955(t1 : \8704(x : U) \8594 sigma x p \8594 p x)\n33\9474           \8594 t1 Omega (\955(x : U) \8594 t1 (tau (sigma x)))\n34\9474 \n35\9474 in  let lem3 : D = \955(p : pow U) \8594 lem1 (\955(y : U) \8594 p (tau (sigma y)))\n36\9474 \n37\9474 in  let lem2\n38\9474         : not D\n39\9474         = lem1\n40\9474           Delta\n41\9474           (   \955(x : U)\n42\9474             \8594 \955(H2 : sigma x Delta)\n43\9474             \8594 \955(H3 : \8704(p : pow U) \8594 sigma x p \8594 p (tau (sigma x)))\n44\9474             \8594 H3 Delta H2 (\955(pp : pow U) \8594 H3 (\955(y : U) \8594 pp (tau (sigma y))))\n45\9474           )\n46\9474 \n47\9474 in  let evidence : bottom = lem2 lem3\n48\9474 \n49\9474 in  evidence\n\nhurkensParadox.dhall:7:5", _relatedInformation = Nothing}]
      should produce correct diagnostic for various typecheck errors:        FAIL
        expected: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 0}, _end = Position {_line = 1, _character = 0}}, _severity = Just DsError, _code = Nothing, _source = Just "dhall-lsp-server", _message = "Alternative annotation mismatch", _relatedInformation = Nothing}]
         but got: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 0}, _end = Position {_line = 0, _character = 33}}, _severity = Just DsError, _code = Nothing, _source = Just "Dhall.TypeCheck", _message = "\n\ESC[1;31mError\ESC[0m: Alternative annotation mismatch\n\n1\9474 < Left : Natural | Right : Type >\n\nmixedUnions.dhall:1:1", _relatedInformation = Nothing}]
      should produce correct diagnostic for various typecheck errors:        FAIL
        expected: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 0}, _end = Position {_line = 1, _character = 0}}, _severity = Just DsError, _code = Nothing, _source = Just "dhall-lsp-server", _message = "Record mismatch", _relatedInformation = Nothing}]
         but got: [Diagnostic {_range = Range {_start = Position {_line = 0, _character = 0}, _end = Position {_line = 0, _character = 28}}, _severity = Just DsError, _code = Nothing, _source = Just "Dhall.TypeCheck", _message = "\n\ESC[1;31mError\ESC[0m: Record mismatch\n\n1\9474 { foo = 1 } \11005 { bar = Text }\n\npreferMixedRecords.dhall:1:1", _relatedInformation = Nothing}]

8 out of 9 tests failed (0.06s)
```

How about we disable the lsp-server for now?

File: appveyor.yml [hunk 1]
```diff
@@ -54,6 +54,14 @@ for:
...
     - 7z a "bin\dhall-bash-%DEPLOY_SUFFIX%" "%APPVEYOR_BUILD_FOLDER%\bin\dhall-to-bash.exe"
     - 7z a "bin\dhall-lsp-server-%DEPLOY_SUFFIX%" "%APPVEYOR_BUILD_FOLDER%\bin\dhall-lsp-server.exe"
+
+  # stack is not able to build test dependencies with lts-6
+  test_script:
+    - stack test dhall:tasty
+    - stack test dhall-json
+    - stack test dhall-text
+    - stack test dhall-bash
+    - stack test dhall-lsp-server
```
Comment (sjakobi) [inline appveyor.yml]: If @Gabriel439 agrees:

```suggestion
    # - stack test dhall-lsp-server # Disabled while the tests are broken.
```

Comment (Gabriel439) [reply appveyor.yml]: Yeah, I think it's fine to disable these tests for now

**Pull request merged**